src/dpq2/conv/from_d_types.d

```diff
@@ -31,7 +31,15 @@ if (is(T == Nullable!R, R))
 Value toValue(T)(T v)
 if(isNumeric!(T))
 {
-    return Value(v.nativeToBigEndian.dup, detectOidTypeFromNative!T, false, ValueFormat.BINARY);
+    static if (is(T == enum))
+    {
+        alias OType = OriginalType!T;
+        return Value((cast(OType)v).nativeToBigEndian.dup, detectOidTypeFromNative!OType, false, ValueFormat.BINARY);
```

Comment:
What if this enum will be of type `string`?
What if this enum will be of type `long` (instead of PG's enum size int4==int)?

Reply:
Added support for string based enums too and integration tests for mentioned cases

Reply:
Looks like we can not support string enums at all, because PG isn't accepts string representation of enum value. We can support only int enums (It is need only check for int type of enum value)

Reply:
Have you looked here? https://github.com/denizzzka/dpq2/pull/94/files#diff-dd704879b650d079419a99c89f3487bcR128

String enum can be easily handled as text.
PG even supports to define custom enum type with values as string by:
`CREATE TYPE foo AS ENUM('bar','baz')`

And with this change you can pass the string enums there just fine.
Only problem is that when reading these values, they will have an unknown OidType so one will need to read them as `::text` but it can definitely work.
Or we can special case the reading of string enums to ignore the unknown OidType for that to work, but I'm not sure about that.

We can also make it that all enums will be send as text values (by their name) assuming that user will create appropriate enum type in PG or relevant checks on the column for the enum values.
And if anyone wants to store enums in the numeric way, it can be done as before by for example `Foo.baz.to!int.toValue` and readed as `v.as!int.to!Foo`

Reply:
I'm intuitive against special cases and text values. If anyone need it can add ::text to return values.